manifests/candlepin.pp

```diff
@@ -68,8 +70,14 @@
     build_dir     => $certs::ssl_build_dir,
   }
 
-  $keystore_password = extlib::cache_data('foreman_cache_data', $keystore_password_file, extlib::random_password(32))
-  $truststore_password = extlib::cache_data('foreman_cache_data', $truststore_password_file, extlib::random_password(32))
+  # Generate and cache the password on the master once
+  # In multi-puppetmaster setups, the user should specify their own
+  $final_keystore_password = pick(
+    $keystore_password, extlib::cache_data('foreman_cache_data', $keystore_password_file, extlib::random_password(32))
```

Comment:
This may be a slight nuance, but in this case I do think the data is generated and cached, even if unused. Would it make sense to use a more traditional approach with `if/else` here?

Reply:
Changed these 2 lines into `if/else` blocks